### DISCUSSION TITLE
How to connect to MongoDB with TLS/SSL?

### DISCUSSION THREAD

**Discussion by jnsjunior** [SoftInstigate/restheart#424] [Q&A] [answered]
Hello,

I want to connect to a MongoDB server that is secured by TLS/SSL.
How can i achieve this?
This is supported by RestHeart?

**Comment by jnsjunior** [comment, marked as answer]
Ok, a found this.
https://restheart.org/docs/security/secure-connection-to-mongodb/

**Reply by mkjsix** [reply]
Beware that paragraph is more about securing RESTHeart under HTTPS, so it's the connection from an HTTP client to RESTHeart itself. If you mean to configure a new MongoDB instance to support TLS/SSL you might want to read this instead: https://docs.mongodb.com/manual/tutorial/configure-ssl/
Then it's just a matter of giving RESTHeart the right `mongo-uri` (see Configuration files section at https://restheart.org/docs/setup/#configuration-files)

**Reply by mkjsix** [reply]
Have also a look at https://docs.mongodb.com/manual/reference/connection-string/#connection-string-options 

Basically, there's nothing special to do in RESTHeart if you want to connect securely to MongoDB, if the database is properly configured to run under TLS|SSL, it is just a matter of using the right `mongo-uri` string.

**Reply by ujibang** [reply]
If MongoDB uses a self signed certificate (or a certificate issued by an authority that isn't in the java `cacerts`), you need to import it into a keystore and pass it to RESTHeart.  

This is explained in the mentioned doc page in section [Connect to MongoDB over TLS](https://restheart.org/docs/security/secure-connection-to-mongodb/#connect-to-mongodb-over-tls)
